File: event-service/models/eventModel.js
import { Schema, model } from "mongoose";

const EventSchema = new Schema({
  title: {
    type: String,
    required: true
  },
  deadline: Date,
  link: {
    type: String,
    required: true
  },
  venue: String,
  remarks: String
});

const Event = model("Event", EventSchema);

export default Event;

File: event-service/models/event_model.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type Event struct {
	gorm.Model
	ID          uuid.UUID `gorm:"primaryKey;type:uuid;default:gen_random_uuid()"`
	Name        string    `gorm:"not null"`
	Description string    `gorm:"not null"`
	Venue       string
	Url         string
	Dates       []time.Time `gorm:"type:timestamptz[]"`
	StartDate   time.Time
	EndDate     time.Time
	Deadline    time.Time
}
